### PR TITLE
Title and subtitle

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: .github/workflows/test-and-build.yml [hunk 1]
```diff
@@ -43,32 +43,15 @@ jobs:
...
-          username: ${{ secrets.HARBOR_USERNAME }}
-          password: ${{ secrets.HARBOR_PASSWORD }}
-
-      - name: Build docker image
-        uses: docker/build-push-action@v2
-        with:
-          registry: registry.cern.ch
-          push: true
-          tags: registry.cern.ch/cern-sis/workflows:latest,registry.cern.ch/cern-sis/workflows:${{ github.sha }}
+    uses: cern-sis/gh-workflows/.github/workflows/build-image.yml@main
```
Comment (ErnestaP) [inline .github/workflows/test-and-build.yml]: Jimil's changes

File: dags/iop/parser.py [hunk 2]
```diff
@@ -51,6 +55,18 @@ def __init__(self) -> None:
...
+                destination="title",
+                required=False,
+                source="front/article-meta/title-group/article-title",
+                extra_function=lambda x: " ".join(x.split()),
+            ),
+            TextExtractor(
+                destination="subtitle",
+                required=False,
+                source="front/article-meta/subtitle",
+                extra_function=lambda x: " ".join(x.split()),
```
Comment (ErnestaP) [inline dags/iop/parser.py:111]: Wanted to remove all double spaces, new lines and etc.

File: dags/common/parsing/xml_extractors.py [hunk 3]
```diff
@@ -1,10 +1,68 @@
...
+            ]
+            extracted_value = "".join(
+                values_in_regular_tags + values_math_expression_and_styling_tags
+            )
+            return WHITE_SPACES.sub("", extracted_value)
+        except AttributeError:
+            self.logger.error(f"{self.destination} is not found in XML")
+            return
+
+    def _process_text_with_extra_function(self, text):
```
Comment (ErnestaP) [inline dags/common/parsing/xml_extractors.py]: _process_text_with_extra_function, extract ar the same in ContentExtractor and in TextExtractor. Should I move these functions to a common file and reuse it here? O better to leave, that code would be more easily readable?

File: dags/common/constants.py [hunk 4]
```diff
@@ -2,3 +2,4 @@
 
 ARXIV_EXTRACTION_PATTERN = re.compile(r"(arxiv:|v[0-9]$)", flags=re.I)
 NODE_ATTRIBUTE_NOT_FOUND_ERRORS = (AttributeError, TypeError)
+WHITE_SPACES = re.compile(r"[\n\t]{0,}" + r"\s{2,}")
```
Comment (MJedr) [inline dags/common/constants.py]: why it can't be one string?

File: dags/common/parsing/xml_extractors.py (hunk 3 above)
Comment (MJedr) [reply dags/common/parsing/xml_extractors.py]: hm maybe you can move it to `IExtractor`? It won't be a super-pure interface (do I assume correctly that I come from an interface?), but well, firstly, it's not Java. Or maybe if you don't want to break Java good practices do a class that will implement this interface and add there this method.

File: dags/common/parsing/xml_extractors.py [hunk 5]
```diff
@@ -1,10 +1,68 @@
 import xml.etree.ElementTree as ET
 
+from common.constants import WHITE_SPACES
 from common.parsing.extractor import IExtractor
 from common.utils import check_value
 from structlog import get_logger
 
 
+class ContentExtractor(IExtractor):
```
Comment (MJedr) [inline dags/common/parsing/xml_extractors.py]: where do you use this extractor?

Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py]: I will use it it Abstracts, if we will keep the regex idea

File: dags/common/constants.py (hunk 4 above)
Comment (ErnestaP) [reply dags/common/constants.py]: + It is "AND" operator:
https://blog.finxter.com/python-regex-and-operator-tutorial-video/

File: dags/common/parsing/xml_extractors.py [hunk 6]
```diff
@@ -14,14 +15,17 @@ def __init__(
         default_value=None,
         extra_function=lambda s: s,
         prefixes=None,
+        all_content_between_tags=False,
```
Comment (MJedr) [inline dags/common/parsing/xml_extractors.py]: I would prefer to change the name of this attribute to something like `with_tags_conent` or `extract_tags_content`

File: dags/common/parsing/xml_extractors.py (hunk 3 above)
Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py]: Removed the Content extractor, extraction function added to TextExtractr. Now, when we want to extract everything between tags, we will need to pass flag in text_extractor:
`all_content_between_tags=True`

File: dags/common/parsing/xml_extractors.py [hunk 7]
```diff
@@ -31,6 +35,20 @@ def _get_text_value(self, raw_value):
...
+    def _get_content_as_text_value(self, node):
+        try:
+            values_in_regular_tags = [node.text or ""]
+            values_math_expression_and_styling_tags = [
+                ET.tostring(el).decode("ascii") for el in node
+            ]
+            extracted_value = "".join(
+                values_in_regular_tags + values_math_expression_and_styling_tags
+            )
+            return WHITE_SPACES.sub("", extracted_value)
```
Comment (MJedr) [inline dags/common/parsing/xml_extractors.py]: can you add a test that will cover this case?

Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py]: actually, we need it just on abstract branch, removing from this one

File: dags/common/parsing/xml_extractors.py (hunk 6 above)
Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py]: actually, we need it just on abstract branch, removing from this one

Comment (MJedr) [reply dags/common/parsing/xml_extractors.py]: you didn't remove in the end ;)

Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py]: sorry, no it's completely removed

**Review by MJedr — APPROVED** [review]

File: dags/iop/parser.py (hunk 2 above)
Comment (drjova) [reply dags/iop/parser.py:111]: do we have such cases? why we need to handle that?

**Pull request merged**